### PR TITLE
Add definition for AWS Lambda Code Pipeline event

### PR DESCRIPTION
**Pull request by nwholloway**
This add a definition for CodePipelineEvent which is the event used with invocations from AWS Code Pipelines.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://docs.aws.amazon.com/codepipeline/latest/userguide/actions-invoke-lambda-function.html
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/aws-lambda/index.d.ts
Comment (simonbuchan) [inline types/aws-lambda/index.d.ts]: Presumably this would become a union of other artifact location types? Just checked [the docs](https://docs.aws.amazon.com/codepipeline/latest/APIReference/API_ArtifactLocation.html), and it does only have the `S3` type, weird. As is it's a little confusing, could maybe use a comment for whoever follows up.

File: types/aws-lambda/index.d.ts
Comment (simonbuchan) [inline types/aws-lambda/index.d.ts]: Ugh, thanks AWS.

File: types/aws-lambda/index.d.ts
Comment (simonbuchan) [inline types/aws-lambda/index.d.ts]: These names are a little generic maybe? I'd prefer `CodePipelineArtifact` etc, but I'm not sure what the `S3Artifact*` stuff should be named.
`Credentials` is fine, as it's a generic type (and I'm suprised we don't already have it)

File: types/aws-lambda/index.d.ts
Comment (simonbuchan) [inline types/aws-lambda/index.d.ts]: From the [JobData](https://docs.aws.amazon.com/codepipeline/latest/APIReference/API_JobData.html) docs, looks like you're missing `encryptionKey` (remembering the lambda event docs call out `actionTypeId` and `pipelineContext` as not included)

File: types/aws-lambda/index.d.ts
Comment (simonbuchan) [reply types/aws-lambda/index.d.ts]: [Artifact](https://docs.aws.amazon.com/codepipeline/latest/APIReference/API_Artifact.html) docs mark all these properties as optional? I don't know how that works.

**Review by simonbuchan — APPROVED** [review]
Looks great!

**Comment by rbuckton** [comment]
@simonbuchan, based on your comments are there changes that you would like to see made before this is merged?

**Comment by simonbuchan** [comment]
I tend to approve even if I note changes I would like to see if I would still be fine with it merged as-is. The most serious issue is the missing `encryptionKey` property, but that probably isn't used too much?

File: types/aws-lambda/index.d.ts
Comment (nwholloway) [reply types/aws-lambda/index.d.ts]: I'll leave the names of the types as is -- they do actually line up with the API reference.  I've left them as mandatory, as I always saw them provided.

File: types/aws-lambda/index.d.ts
Comment (nwholloway) [reply types/aws-lambda/index.d.ts]: I'll add `encryptionKey`.  It wasn't something I'd seen, but I wasn't using encrypted artifacts.

**Review by rbuckton — APPROVED** [review]

**Comment by rbuckton** [comment]
Thanks for the contribution!

**Pull request merged**
